model/project_ref.go

```diff
@@ -607,6 +609,42 @@ func (p *ProjectRef) GetBatchTime(variant *BuildVariant) int {
 	}
 }
 
+// return the next valid batch time
+func GetActivationTimeWithCron(curTime time.Time, cronBatchTime string) (time.Time, error) {
+	parser := cron.NewParser(cron.Hour | cron.Dom | cron.DowOptional | cron.Descriptor)
```

Comment:
I would think it would make sense to do this in the same way as in `man 5 crontab`, since that's what people are used to: `NewParser(Minute | Hour | Dom | Month | Dow)`. That way people know what they can specify without reading docs. This also allows people to specify something not on the hour. Including `cron.Descriptor` seems reasonable too.